aws_vpc_policies/aws_security_group_administrative_ingress.py

```diff
@@ -14,21 +14,16 @@ def port_checker(check_ports, from_port, to_port):
 
 
 def policy(resource):
-    if resource['SecurityGroups'] is None:
+    if resource['IpPermissions'] is None:
         return True
 
-    for group in resource['SecurityGroups']:
-        if group['IpPermissions'] is None:
-            continue
-
-        for permission in group['IpPermissions']:
-            src_open = False
-            for ip_range in permission['IpRanges'] or []:
-                src_open = src_open or ip_range['CidrIp'] == '0.0.0.0/0'
-            for ipv6_range in permission['Ipv6Ranges'] or []:
-                src_open = src_open or ipv6_range['CidrIpv6'] == '::/0'
-            if src_open and port_checker(RESTRICTED_PORTS,
-                                         permission['FromPort'],
-                                         permission['ToPort']):
-                return False
+    for permission in resource['IpPermissions']:
+        src_open = False
+        for ip_range in permission['IpRanges'] or []:
+            src_open = src_open or ip_range['CidrIp'] == '0.0.0.0/0'
+        for ipv6_range in permission['Ipv6Ranges'] or []:
+            src_open = src_open or ipv6_range['CidrIpv6'] == '::/0'
```

Comment:
Could we break out of these for loops as soon as src_open is true? I've seen iptables rules get pretty long per port, but I don't know how short/long _these_ lists could be, so perhaps doing so would be negligible.

Reply:
Yeah seems reasonable.